### PR TITLE
Add BRANCH_KEY to configure branch.io banners.

### PR DESCRIPTION
**Pull request by waheedahmed**
LEARNER-2731

edx-paltform PR: https://github.com/edx/edx-platform/pull/16823

Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [x] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

**Comment by waheedahmed** [comment]
@fredsmith, @feanil Can I get a quick review for this PR please?

File: CHANGELOG.md [hunk 1]
```diff
@@ -474,3 +474,6 @@
 
 - Role: edxapp
   - Added `EDXAPP_ZENDESK_OAUTH_ACCESS_TOKEN` for making requests to Zendesk through front-end.
+
```
Comment (feanil) [inline CHANGELOG.md]: Changelog entries should go at the top of the file. not at the bottom.

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -967,6 +968,7 @@ edxapp_generic_auth_config:  &edxapp_generic_auth
   FACEBOOK_APP_SECRET: "{{ EDXAPP_FACEBOOK_APP_SECRET }}"
   FACEBOOK_API_VERSION: "{{ EDXAPP_FACEBOOK_API_VERSION }}"
   ENTERPRISE_SERVICE_WORKER_USERNAME: "{{ EDXAPP_ENTERPRISE_SERVICE_WORKER_USERNAME }}"
+  BRANCH_KEY: "{{ EDXAPP_BRANCH_KEY }}"
```
Comment (feanil) [inline playbooks/roles/edxapp/defaults/main.yml]: `BRANCH_KEY` is a very generic name, please change it to `BRANCH_IO_KEY` and please change `EDXAPP_BRANCH_KEY` to `EDXAPP_BRANCH_IO_KEY`

**Review by feanil — CHANGES_REQUESTED** [review]
Can you also link to the edx-platform PR that integrates this setting into the platform?

**Comment by waheedahmed** [comment]
@feanil made the suggested changes and here is the platform PR https://github.com/edx/edx-platform/pull/16823, kindly take another look.

**Review by feanil — APPROVED** [review]

**Pull request merged**